Kudu.Core/Jobs/JobsManagerBase.cs

```diff
@@ -77,9 +83,49 @@ protected JobsManagerBase(ITraceFactory traceFactory, IEnvironment environment,
             JobsDataPath = Path.Combine(Environment.JobsDataPath, jobsTypePath);
 
             HostingEnvironment.RegisterObject(this);
+            JobsFileWatcher = new JobsFileWatcher(JobsBinariesPath, OnJobChanged, JobSettings.JobSettingsFileName, ListJobNames, TraceFactory, Analytics);
         }
 
-        public abstract IEnumerable<TJob> ListJobs();
+        /// <summary>
+        /// Process job file changed and invoke any extra file watcher event handlers
+        /// </summary>
+        private void OnJobChanged(string jobName)
+        {
+            JobListCache = null;
+            if (FileWatcherExtraEventHandlers != null)
+            {
+                foreach (Action<string> action in FileWatcherExtraEventHandlers)
+                {
+                    action.Invoke(jobName);
```

Comment:
Instead of EventHandlers/Callback model, I wonder if we can make OnJobChanged protected virtual - derived class just override it?